analyzers/tests/SonarAnalyzer.UnitTest/TestCases/GenericTypeParametersRequired.cs

```diff
@@ -56,17 +56,8 @@ public static T MyMethod_23<T>(string value)
         public void MyMethod_24<T, TBase>(T arg) where T : TBase { }
         public void MyMethod_25<T, TBase>(T arg) where T : IEnumerable<TBase> { }
         public void MyMethod_26<T, TBase>(T arg) where T : IEnumerable<int>, TBase { }
-        public TypeMap MyMethod_271<T, TBase>() => new TypeMap(typeof(T), typeof(TBase)); // Noncompliant FP
```

Comment:
do we already have this case in another place in the file? otherwise, I would keep it as it's a good regression detection scenario

Reply:
This is not on the master branch, I added it to my feature branch when I was trying to run the integration tests to see if the unit tests behave differently or not.
I also looked at other rules, and I'm not sure whether this is really an FP.

Reply:
Not as an FP, but as a test case, to document the behavior, to know later what we think of it: FP or not.

Tests are the best documentation

Reply:
True, but I just checked the rules, other than [S4018](https://rules.sonarsource.com/csharp/RSPEC-4018) there's also [S2326](https://rules.sonarsource.com/csharp/RSPEC-2326), which actually covers this.
[S4018](https://rules.sonarsource.com/csharp/RSPEC-4018) is about type inference, and it only checks the method declaration, while [S2326](https://rules.sonarsource.com/csharp/RSPEC-2326) checks if it's used in the method body.
So I'm somewhat puzzled by how to handle this. 

There are methods in DI containers that use generic type parameters:

```
public RegisterOptions Register<RegisterType>()
{
    return this.Register(typeof(RegisterType));
}
```

This one cannot be called without explicitly stating the type parameter (so type inference cannot be used), but I still wouldn't issue a warning for it.

Reply:
Update: we should not deprecate the rule there can be some teams that don't want diamond operators `<>` in their method invocations (for some reason) and only want to use type inference, so they might design their whole codebase to respect that.